src/app/zeko/circuits/indexed_merkle_tree.mli

```diff
@@ -27,7 +27,7 @@ module Make (Inputs : sig
 end) : sig
   open Inputs
 
-  type t
+  type t = F.t
```

Comment:
Needed to construct the type

Reply:
You mean this?
```
let to_account_set x =
  let (Typ typ) = Account_set.typ in
  typ.value_of_fields ([| x |], typ.constraint_system_auxiliary ())
```

How is this good? When you will change the type it won't give any compilation error and will leave me wondering what's wrong. If you want I can add a function `of_field` but converting from fields is not a good solution imho.

Reply:
I would hope it moves somewhere else, not disappear. When we will be transitioning it will be exceptionally hard to upgrade the code without type checks.

Reply:
I've changed this for the sake of merging the PR, but I still think that bypassing typechecks is a bad idea that might bite us in the future.

Reply:
I think removing type checks should be done at the same time as adding guard rails for constructing the witnesses correctly.

Reply:
Yes there will be some guardrails.

Reply:
They should be added at the same time as we are removing guardrails of type checking